src/lcm/dispatchers.py

```diff
@@ -191,21 +191,81 @@ def allow_kwargs(func):
 
     @functools.wraps(func)
     def allow_kwargs_wrapper(*args, **kwargs):
-        parameters = list(inspect.signature(func).parameters)
-
-        positional = list(args) if args is not None else []
-
-        kwargs = {} if args is None else kwargs
-        if len(args) + len(kwargs) != len(parameters):
+        parameters = inspect.signature(func).parameters
+
+        # Retrieve keyword-only arguments and remove from kwargs
+        keyword_only = {
+            k: v
+            for k, v in kwargs.items()
+            if k in parameters and parameters[k].kind == inspect.Parameter.KEYWORD_ONLY
```

Comment:
Instead of converting all but the KEYWORD_ONLY arguments into positional ones, we could just convert the POSITIONAL_ONLY ones

Reply:
After discussion we put less focus on style and just try to work around all problem introduced by vmap. Example: 

```python
from jax import vmap
import jax.numpy as jnp
import inspect

def f(*, a, b):
    return a + b


v = vmap(f, in_axes=(0, 0))

parameters = inspect.signature(v).parameters

parameters["a"].kind
```
```txt
<_ParameterKind.KEYWORD_ONLY: 3>
```
I.e., despite being a positional only parameter (due to jax.vmap), the signature still says that we have a keyword only argument here.